torch/utils/data/dataloader.py

```diff
@@ -1489,17 +1497,25 @@ def _try_put_index(self):
         for _ in range(self._num_workers):  # find the next active worker, if any
             worker_queue_idx = next(self._worker_queue_idx_cycle)
             if self._workers_status[worker_queue_idx]:
-                break
+                if self._in_order:
+                    break
+                elif self._workers_num_tasks[worker_queue_idx] < max_tasks // sum(
```

Comment:
let's make a note here that workers_status is only updated in this thread, so there is no chance of division-by-zero

Reply:
Good call - done.
It's also guaranteed > 0 since this whole block sits within `if self._workers_status[worker_queue_idx]:` on L1499 so it has to be at least 1 in order for the sum to be computed